tests/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,351 @@
+    @Warmup(global = true)
+    public void warmup() throws InterruptedException {
+        Random random = new Random();
+        for (int k = 0; k < keyCount; k++) {
+            int id = random.nextInt();
+            String key = StringUtils.generateString(keyLength);
+            EmployeeInterface value = createEmployee(id);
+            map.put(key, value);
+        }
+        log.info("Map size is:" + map.size());
+    }
+
+    @Run
+    public void run() {
+        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
+        for (int k = 0; k < threadCount; k++) {
+            spawner.spawn(new Worker());
+        }
+        spawner.awaitCompletion();
+    }
+
+    @Performance
+    public long getOperationCount() {
+        return operations.get();
+    }
+
+    private class Worker implements Runnable {
+        @Override
+        public void run() {
+            long iteration = 0;
+            Metronome metronome = SimpleMetronome.withFixedIntervalMs(intervalMs);
+            SqlPredicate sqlPredicate = new SqlPredicate(sqlQuery);
+            NamePredicate namePredicate = new NamePredicate("aaa");
+            while (!testContext.isStopped()) {
+                metronome.waitForNext();
+                search.started();
+                if (customPredicate) {
+                    map.values(namePredicate);
+                } else {
+                    map.values(sqlPredicate);
+                }
+                search.done();
+
+                iteration++;
+                if (iteration % logFrequency == 0) {
+                    log.info(Thread.currentThread().getName() + " At iteration: " + iteration);
+                }
+
+                if (iteration % performanceUpdateFrequency == 0) {
+                    operations.addAndGet(performanceUpdateFrequency);
+                }
+
+            }
+            operations.addAndGet(iteration % performanceUpdateFrequency);
+        }
+    }
+
+    public EmployeeInterface createEmployee(int id) {
+        switch (serializationStrategy) {
+            case DATA_SERIALIZABLE:
+                return new DataSerializableEmployee(id);
+            case JAVA_SERIALIZABLE:
+                return new SerializableEmployee(id);
+            case IDENTIFIED_DATA_SERIALIZABLE:
+                return new IdentifiedDataSerializableEmployee(id);
+            case PORTABLE:
+                return new PortableEmployee(id);
+        }
+        return null;
+    }
+
+    //Custom Predicates
+    private static class NamePredicate implements Predicate<String, EmployeeInterface> {
+        private final String name;
+
+        private NamePredicate(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean apply(Map.Entry<String, EmployeeInterface> entry) {
+            EmployeeInterface value = entry.getValue();
+            return value.getName().equals(name);
+        }
+    }
+
+    public interface EmployeeInterface {
```

Comment:
Don't call an interface 'FooInterface'. Just call it Foo, in this case Employee. Adding interface to it is totally not informative or relevant.